frontend/src/metabase/query_builder/components/notebook/steps/DataStep/DataStep.tsx

```diff
@@ -65,6 +66,14 @@ export const DataStep = ({
     // we need to populate question metadata with selected table
     await dispatch(Tables.actions.fetchMetadata({ id: tableId }));
 
+    if (typeof tableId === "string") {
```

Comment:
Porting [this change](https://github.com/metabase/metabase/pull/42330/files#r1596530607) to this component as well.